### DISCUSSION TITLE
Migrating from NB to OS

### DISCUSSION THREAD

**Discussion by tedkrapf** [openstore-ecommerce/OpenStore#193] [Q&A]
Hi team,

I have an existing NB Store site that is now broken due to upgrading to DNN 9x.  Per the available instructions/discussions, I installed the OpenSource (and templating) modules, and got the OS BO up and running.  I thought I'd be able to go directly into the database and write a few queries to migrate the prod/cat/img/etc data from the NB to OS tables, but obviously discovered that OS using the new single/xml table (cool design, by the way).  

Since doing a SQL query data migration looked incredibly cumbersome, my next thought was to add a single product/cat/img/etc using the OS BO, run an export, check out that file format, and then SQL query output my NB data to that OS import format.  But then I saw that the single product/cat/img OS export is over 1k lines of XML -- which seems like a LOT of work to put all of my NB products into that format so I can use the OS import feature.

- Is there an easier way to migrate NB data to OS?
- Has anyone already done the leg work of producing an SQL query that outputs the OS XML format from the original NB tables?
- Any other advice you can offer?

THANKS!!

**Comment by leedavi** [comment]
Hi, sorry about the late reply.

What version of NBStore did you have?

NBS v3 should work with export and import into OpenStore v4.  The DB is nearly identical.

https://github.com/openstore-ecommerce/OpenStore/blob/master/Installation/Readme_NBS_Upgrade.txt

NBSv2 is another problem, the DB changed.  You need to import into NBSv3 first and then into OpenStorev4.

You can use this to migrate v2 to v3 
https://github.com/FabioParigi/NBrightBuyMigrate

**Comment by tedkrapf** [comment]
Thanks @leedavi for the info.
No worries on the delayed response.

It was from NB v2.x.  Fortunately it was only about 40 products with about 100 models total.  Since it was time sensitive, I just dumped all the NB v2 tables to a few Excel docs, grabbed all the product images, and gave all the info to one of my underlings and they spent a few hours entering everything back into the new OS BO while I setup of the modules and styling.

Keep up the great work with OS, it's looks like a tremendous step forward from the already great NB!

All the best.
